rotkehlchen/globaldb/handler.py

```diff
@@ -2013,31 +2017,56 @@ def get_assetid_from_exchange_name(exchange: Location | None, symbol: str, defau
         return default if identifier is None else identifier[0]
 
     @staticmethod
-    def query_location_asset_mappings(
-            filter_query: 'LocationAssetMappingsFilterQuery',
+    def query_asset_mappings_by_type(
+            mapping_type: Literal['location', 'counterparty'],
```

Comment:
This is not a well abstracted function. It's full of if conditions even inside a for loop!!!

Either add more arguments to abstract it or spit it into two functions with some common logic.

Pass the columns as an argument, pass the mapping_keys as an argument and pass the location_or_counterparty_reader_callback as an argument. That's if you want to keep this into this one abstracted function.